cyral/resource_cyral_sidecar.go

```diff
@@ -97,13 +97,21 @@ func resourceSidecarRead(ctx context.Context, d *schema.ResourceData, m interfac
 
 	body, err := c.DoRequest(url, http.MethodGet, nil)
 	if err != nil {
-		return createError(fmt.Sprintf("Unable to read sidecar. SidecarID: %s",
-			d.Id()), fmt.Sprintf("%v", err))
+		// Currently, the sidecar API always returns a status code of 500 for every error,
+		// so its not possible to distinguish if the error returned is related to
+		// a 404 Not Found or not. This way, once this is fixed in the sidecar API,
+		// we should handle the error here by its status code, and only remove the
+		// resource from the state (d.SetId("")) if it returns a 404 Not Found,
+		// otherwise we should return the error in the Diagnostics.
+		log.Printf("[DEBUG] Unable to read sidecar. SidecarID: %s. "+
+			"Removing it from state. Error: %v", d.Id(), err)
+		d.SetId("")
```

Comment:
If the configuration for the control plane is incorrect, I imagine there will also be an error there, as well as with connection issues and etc. I don't think just checking if there was an error is enough, we need to know if the error is actually a "does not exist" error before removing it from the state, otherwise there will be side effects in the future.

Reply:
That makes sense, although we are not able to check the not found error by its status code, so the only workaround solution I see is to check if the error message matches the `Failed to
extract info for wrapper` message, which is returned when the sidecar is not found. Otherwise, we will have to fix it in the sidecar API.

Reply:
@VictorGFM , I related this ticket internally in Jira to the one that will fix the API.